### PR TITLE
PKCondition: added tests for subexpr inference (#951)

### PR DESCRIPTION
**Pull request by vavrusa**
These tests should prove that primary key containing subexpression was used.

### CODE REVIEW

**Comment by vavrusa** [comment]
Hi 👋, I got back from vacation, sorry it took so long. Is the test suite going to be run by CI on this PR?

File: dbms/tests/queries/0_stateless/00486_pk_subexpression.sql [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+SET max_rows_to_read = 5;
+SELECT * FROM test.pk WHERE x = toDateTime(1);
+
+-- Index works on interval 00:01:00 - 00:01:59
+SET max_rows_to_read = 4;
+SELECT * FROM test.pk WHERE x BETWEEN toDateTime(60) AND toDateTime(119) AND y = 11;
+
+-- Cannot read less rows as PK is coarser on interval 00:01:00 - 00:02:00
+SET max_rows_to_read = 5;
+SELECT * FROM test.pk WHERE x BETWEEN toDateTime(60) AND toDateTime(120) AND y = 11;
```
Comment (vavrusa) [inline dbms/tests/queries/0_stateless/00486_pk_subexpression.sql]: ^ I'm not sure how to make a test that would prove that such query cannot work with `max_rows_to_read = 4`.

**Comment by robot-metrika-test** [comment]
Can one of the admins verify this patch?

**Comment by robot-metrika-test** [comment]
Can one of the admins verify this patch?

**Comment by alexey-milovidov** [comment]
autotests

File: dbms/tests/queries/0_stateless/00486_pk_subexpression.sql (hunk 1 above)
Comment (alexey-milovidov) [inline dbms/tests/queries/0_stateless/00486_pk_subexpression.sql]: Simple `sql` tests are capable of testing only "successful" behaviour.
For testing exceptions, it is possible to use `sh` tests, but it is not worth to do in this case.

**Comment by alexey-milovidov** [comment]
Thanks!

**Comment by alexey-milovidov** [comment]
> Is the test suite going to be run by CI on this PR?

Yes. In fact, our CI is:
- internal (results are not published);
- broken at half the time;

File: dbms/tests/queries/0_stateless/00486_pk_subexpression.reference [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+2000-01-01      1970-01-01 00:02:08     55      1235
+2000-01-01      1970-01-01 00:02:07     55      2791
+2000-01-01      1970-01-01 00:02:06     55      5786
+2000-01-01      1970-01-01 00:00:01     11      1235
+2000-01-01      1970-01-01 00:01:03     11      3572
+2000-01-01      1970-01-01 00:01:01     11      4563
+2000-01-01      1970-01-01 00:01:02     11      4578
+2000-01-01      1970-01-01 00:01:03     11      3572
+2000-01-01      1970-01-01 00:01:01     11      4563
+2000-01-01      1970-01-01 00:01:02     11      4578
```
Comment (alexey-milovidov) [inline dbms/tests/queries/0_stateless/00486_pk_subexpression.reference]: No newline at end of file, and this differs from query result, that always have newline.

**Review by alexey-milovidov — COMMENTED** [review]
.

File: dbms/tests/queries/0_stateless/00486_pk_subexpression.reference (hunk 2 above)
Comment (alexey-milovidov) [inline dbms/tests/queries/0_stateless/00486_pk_subexpression.reference]: https://github.com/yandex/ClickHouse/commit/864cae7107723cfda5b0b8aff8f94ed5456188cf

**Pull request merged**